src/android/Print.java

```diff
@@ -259,6 +259,31 @@ private boolean createReceiptData(Order order) {
 			if (order.customer_address != null && !order.customer_address.isEmpty())
 				builder.append("Address: " + order.customer_address + "\n");
 
+			if (order.handoff_provider != null && !order.handoff_provider.isEmpty()) {
+				String formatted_provider = "";
+
+				switch (order.handoff_provider) {
+					case "ubereats": formatted_provider = "Uber Eats";
+					break;
+
+					case "postmastes": formatted_provider = "Postmates";
+
+					case "doordash": formatted_provider = "DoorDash";
+					break;
+
+					case "grubhub": formatted_provider = "Grubhub";
+					break;
+
+					case "skip": formatted_provider = "Skip the Dishes";
+					break;
+				}
+
+				builder.append("Marketplace Provider: " + formatted_provider + "\n");
+			}
+
+			if (order.delivery_id != null && !order.delivery_id.isEmpty())
+				builder.append("Marketplace Delivery ID: " + order.delivery_id + "\n");
```

Comment:
If this is for DD Drive, this isn't a marketplace order -- I think you just want to say like "Delivery Provider ID" or just "Delivery ID"?

Example non-marketplace, third-party delivery order:
https://od.pxsweb.com/admin/internal/db/order/64304661b800910deb0d42c4
(should not say "marketplace")

Example marketplace order:
https://od.pxsweb.com/admin/internal/db/order/64304406a7b4b8805c0c1de6
(here, if any ID is printed, I think you'd want to print the handoff_display_id field)